include/COLA.hh

```diff
@@ -195,36 +204,68 @@ namespace cola {
         void regWrite(std::shared_ptr<VFactory>&& factory, const std::string& name){writerMap.emplace(name, std::move(factory));}
-    };
 
-    class ColaRunManager {
-    public:
-        explicit ColaRunManager(FilterAnsamble&& ansamble) : filterAnsamble(std::move(ansamble)) {}
-        ~ColaRunManager() = default;
-        void run();
-        void test();
-    private:
-        FilterAnsamble filterAnsamble;
+        MetaData parseStrToMeta(std::string data);
     };
 
 
-    inline cola::FilterAnsamble cola::MetaProcessor::parse(const cola::MetaData& data) {
+    inline cola::FilterAnsamble cola::MetaProcessor::parse(const cola::SimpleMetaData& data) {
+        FilterAnsamble ansamble;
+        ansamble.generator = std::shared_ptr<VGenerator>(dynamic_cast<VGenerator*>(generatorMap.at(data.generatorName)->create("")));
+        ansamble.converters.push(std::shared_ptr<VConverter>(dynamic_cast<VConverter*>(converterMap.at(data.converterName)->create(""))));
+        ansamble.writer = std::shared_ptr<VWriter>(dynamic_cast<VWriter*>(writerMap.at(data.writerName)->create("")));
+        return ansamble;
+    }
+
+    inline cola::FilterAnsamble cola::MetaProcessor::parse(std::string data) {
         FilterAnsamble ansamble;
-        ansamble.generator = std::shared_ptr<VGenerator>(dynamic_cast<VGenerator*>(generatorMap.at(data.generatorName)->create()));
-        ansamble.converters.push(std::shared_ptr<VConverter>(dynamic_cast<VConverter*>(converterMap.at(data.converterName)->create())));
-        ansamble.writer = std::shared_ptr<VWriter>(dynamic_cast<VWriter*>(writerMap.at(data.writerName)->create()));
+        MetaData meta = parseStrToMeta(data);
+        ansamble.generator = std::shared_ptr<VGenerator>(dynamic_cast<VGenerator*>(generatorMap.at(meta.generatorName)->create(meta.filterParamMap.at(meta.generatorName))));
+        while(!meta.converterNames.empty())
+        {ansamble.converters.push(std::shared_ptr<VConverter>(dynamic_cast<VConverter*>(converterMap.at(meta.converterNames.front())->create(meta.filterParamMap.at(meta.converterNames.front())))));
+            meta.converterNames.pop();}
+        ansamble.writer = std::shared_ptr<VWriter>(dynamic_cast<VWriter*>(writerMap.at(meta.writerName)->create(meta.filterParamMap.at(meta.writerName))));
         return ansamble;
     }
 
+    inline cola::MetaData cola::MetaProcessor::parseStrToMeta(std::string data) {
+        std::vector<std::string> filters; cola::MetaData metaData;
+        boost::split(filters, data, boost::is_any_of("\n"));
+        for(int flt = 0; flt < filters.size(); ++flt){
+            std::vector<std::string> tmp;
+            boost::split(tmp, filters.at(flt), boost::is_any_of(" "));
+            std::string res = "";
+            for(int i = 1; i < tmp.size(); ++i){res+=tmp.at(1);}
+            metaData.filterParamMap.emplace(tmp.at(0),res);
+            if(flt == 0){metaData.generatorName = tmp.at(0);}
```

Comment:
please, remove redundant curved braces for one-line if-else/for statements and add spacing. Or even better, make it indented like this:
```
if (0 < 1)
    std::cout << 1;
```

Reply:
It is a part of the if else statement

Reply:
I consulted Google C++ Style and it insists on using spaces and not using compact version of if-else statements.
![image](https://github.com/Spectator-matter-group-INR-RAS/COLA/assets/67229061/e04c454a-e173-4fff-a148-245f08c3dc38)

Reply:
done